src/pydata_sphinx_theme/assets/scripts/index.js

```diff
@@ -71,7 +79,74 @@ function scrollToActive() {
   });
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Theme interaction
+////////////////////////////////////////////////////////////////////////////////
+var prefersDark = window.matchMedia("(prefers-color-scheme: dark)");
+
+function autoTheme(e) {
+  document.body.dataset.theme = prefersDark.matches ? "dark" : "light";
+}
+
+function setTheme(mode) {
+  if (mode !== "light" && mode !== "dark" && mode !== "auto") {
+    console.error(`Got invalid theme mode: ${mode}. Resetting to auto.`);
+    mode = "auto";
+  }
+
+  // get the theme
+  var colorScheme = prefersDark.matches ? "dark" : "light";
+  document.body.dataset.mode = mode;
+  document.body.dataset.theme = mode == "auto" ? colorScheme : mode;
+
+  // save mode
+  localStorage.setItem("theme", mode);
+  console.log(`Changed to ${mode} mode.`);
+
+  // add a listener if set on auto
+  prefersDark.onchange = mode == "auto" ? autoTheme : "";
+}
+
+function cycleTheme() {
+  const defaultMode = document.body.dataset.defaultMode || "auto";
+  const currentTheme = localStorage.getItem("theme") || defaultMode;
+
+  if (prefersDark.matches) {
```

Comment:
I think you could simplify this with a function that could be re-used like:

```javascript
var loopArray = (arr, current) => {
  var nextPosition = arr.indexOf(current) + 1;
  if (nextPosition === arr.length) {
    nextPosition = 0;
  }
  return arr[nextPosition];
};
```

and then call it like

```javascript
let newTheme = loopArray(["auto", "light", "dark"], currentTheme);
setTheme(newTheme);
```